### PR TITLE
feat(compliance): add devflow-compliance optional plugin (Phases 1+2)

### PR DESCRIPTION
**Pull request by dean0x**
## Summary

Adds the `devflow-compliance` optional plugin, delivering code-level regulatory compliance patterns across GDPR, HIPAA, PCI DSS, SOC 2, ISO 27001, and SOX — integrated into the code-review, plan, and implement pipelines. The plugin is entirely opt-in: without installation, zero compliance surface is exposed and no warnings are emitted.

### Phase 1 — Plugin Foundation (commit `0f3ce88`)

- **`shared/rules/compliance.md`** — Global rule (`paths: []`): 6 data-control directives enforced on every prompt when the plugin is installed
- **`shared/skills/compliance/SKILL.md`** — 137-line skill with Iron Law, 6 control categories (access, encryption, audit, retention/erasure, boundary, IaC), severity/framework mapping tables
- **`shared/skills/compliance/references/`** — 8 reference files: GDPR, HIPAA, PCI DSS, SOC 2, ISO 27001, SOX, detection patterns, sources (200+ regulatory citations)
- **`plugins/devflow-compliance/.claude-plugin/plugin.json`** — Plugin manifest v1.8.3; `skills: ['compliance']`, `rules: ['compliance']`, `optional: true`
- **`src/cli/plugins.ts`** — DEVFLOW_PLUGINS entry; `'compliance'` added to LEGACY_SKILLS_V2X
- **`src/cli/commands/init.ts`** — Plugin hint + Step 2 label update + advanced rules note
- **`.claude-plugin/marketplace.json`** — Marketplace entry

### Phase 2 — Workflow Integration (commit `afb5271`)

- **`commands/_partials/_compliance.mds`** — New installed-gate partial: scope-aware manifest check → skill asset verify → rule-file fallback → CLAUDE.md opt-out. ≤3 file reads, no subprocess, no LLM call, silent (applies ADR-007)
- **`commands/code-review.mds`** — Import + `{compliance_gate()}` in Phase 1; compliance row in REVIEWER_LIST table; reviewer count bounded ≤20 per worktree
- **`commands/plan.mds`** — Import + `{compliance_gate()}` in Phase 5 Gap Analysis; 4→5 Designer agents when `COMPLIANCE_ENABLED`; compliance focus row in gap-analysis table
- **`commands/implement.mds`** — Import + `{compliance_gate()}` in Phase 1 Setup; `COMPLIANCE` field added to ALL Coder prompt templates (initial + validation-fix + alignment-fix + qa-fix loops) for consistent guidance; absent = no-op (backward-compatible with /resolve and /dynamic-build)
- **`shared/agents/reviewer.md`** — `compliance` focus row + conditional activation note with clean-report contract
- **`shared/agents/designer.md`** — `compliance` added to gap-analysis focus list in description and Modes table
- **`shared/agents/synthesizer.md`** — Merge-don't-boost note in both design and review modes: a compliance+security co-located finding is one gap seen by two lenses, not multi-agent corroboration
- **`shared/agents/coder.md`** — `COMPLIANCE` input documented; Step 2 body-instructs `Skill(skill="devflow:compliance")` when `COMPLIANCE: enabled` AND regulated surface detected; never in frontmatter (avoids PF-002 re-entrancy guard)
- **`shared/agents/triager.md`** — One-line compliance disposition hint: default FIX_SEPARATE/TECH_DEBT unless code-local
- **`shared/skills/gap-analysis/SKILL.md`** — §7 Compliance: detection patterns for regulatory gaps security doesn't cover (retention/erasure, audit-trail completeness, segregation of duties, IaC exposure)
- **`tests/build-mds.test.ts`** — Partial count assertion 9→10 (RED→GREEN with new partial)
- **`tests/build.test.ts`** — Generic plugin.json↔DEVFLOW_PLUGINS skills+rules sync test for all 23 plugins (closes the unguarded manual-sync pair)
- **Docs** — CLAUDE.md, README.md, docs/commands.md, file-organization.md, skill-catalog.md: 23 plugins / 41 skills / 13 rules / up to 20 reviewers / 10 partials (applies ADR-003)

## Breaking Changes

None. The plugin is optional. Without installation: `COMPLIANCE_ENABLED=false`, no compliance reviewer/designer/prompt field appears, no warnings emitted (AC-3, AC-13).

## Reviewer Focus Areas

- **`commands/_partials/_compliance.mds`** — Gate logic correctness: scope-aware manifest → asset verify → fallback → opt-out; verify AC-11, AC-13, AC-15
- **`shared/agents/coder.md`** — PF-002 guard: body-instructed `Skill(skill="devflow:compliance")` must NOT appear in frontmatter `skills:` list (verify the two are disjoint)
- **`commands/implement.mds`** — `COMPLIANCE` field coverage: all 6 Coder prompt templates (SINGLE, SEQ phase 1, SEQ phase 2+, PARALLEL × 2, plus 3 fix-loop spawns) should carry the field
- **`tests/build.test.ts`** — New sync test: verifies `plugin.json` skills+rules arrays match `DEVFLOW_PLUGINS` for every plugin; confirm it would catch a real mismatch (drift detection)

### CODE REVIEW

File: src/cli/plugins.ts [hunk 1]
```diff
@@ -541,6 +550,8 @@ const LEGACY_SKILLS_V2X: string[] = [
   'devflow:dream-curation',
   // v3.x agent-teams removal: namespaced name for cleanup of installed devflow:agent-teams skill
   'devflow:agent-teams',
+  // v2.x compliance plugin: bare name for pre-namespace installs
+  'compliance',
```
Comment (dean0x) [inline src/cli/plugins.ts]: not sure I understand why this is not prefixed like `devflow:compliance`, why? as far as I remember this is important for everything to work correctly, for example shadowing functionality and proper name spaced installation etc.

**Comment by dean0x** [comment]
## PR Comments Summary

### Inline Comments Created (≥80% confidence)
- **6 inline comments** created across critical findings in shared/skills/, commands/, docs/, and src/cli/
- All comments include confidence levels, severity, and suggested fixes

### Lower-Confidence Suggestions (60-79%)
These are worth considering but not blocking:

1. **Near-duplicate sync tests in tests/build.test.ts** (65%) — The two new test blocks are ~90% identical (skills vs rules). Could collapse to one `it.each()` to remove duplicated scaffolding. Judgment call — explicit form is also defensibly clearer.

2. **Gate scope-precedence false negative in mixed-scope installs** (68%) — `compliance_gate()` in `commands/_partials/_compliance.mds:13` precedence means home-scope installs are invisible when any local manifest exists. May be by design ("local manifest is authoritative"), but surfaces edge case.

3. **No regression guard for compliance gate wiring presence** (68%) — `build-mds.test.ts` verifies partial discovery but not that compliance gate content actually appears in compiled outputs. A developer who `@import`s the partial but forgets the call site produces valid compile with gate silently absent. Cheap grep test would close this.

4. **No regression guard for PF-002 on coder.md** (70%) — `coder.md` correctly body-instructs the skill (not frontmatter). A lightweight grep test ("no agent's frontmatter lists `devflow:compliance`") would guard against future edits silently reintroducing PF-002.

5. **Inconsistent COMPLIANCE field placement across Coder templates** (62%) — In the 5 primary templates the field sits between `DECISIONS_CONTEXT` and `PR_DESCRIPTION_GUIDANCE`, but in fix templates it lands differently. Purely a scannability nit — no behavioral impact.

6. **Orphaned .tmp file on build-mds.ts rename failure** (65-70%) — If `renameSync` fails after `writeFileSync(tmp)`, the temp file is left on disk. Impact is negligible (self-healing on next run, gitignored), but `try/finally` cleanup would align with ADR-003 (leave-the-end-state).

### Items Not in Diff (Cannot Comment Inline)

**docs/reference/release-process.md:111** — Plugin count left stale by this PR (Confidence: 90%, MEDIUM Should Fix)
- Problem: Line reads "All 22 plugin.json files updated". PR added the 23rd plugin but this reference wasn't updated.
- Fix: Change "22 plugin.json files" → "23 plugin.json files"
- **Action required:** This file is not in the current diff, so update must be made separately.

### Category Breakdown

| Category | Count | Status |
|----------|-------|--------|
| Blocking Issues (≥80%) | 7 files | 6 inline + 1 out-of-diff |
| Should Fix (≥80%) | 1 | ✓ 6 inline comments created |
| Lower Confidence (60-79%) | 6 | Summary noted |
| Pre-existing (out-of-scope) | 1 | CONTRIBUTING.md staleness noted in reviews |

### Review Scores Summary
- **Security**: 10/10 — APPROVED
- **Consistency**: 8/10 — APPROVED_WITH_CONDITIONS (6 items fixed via comments)
- **Documentation**: 8/10 — APPROVED_WITH_CONDITIONS (items noted)
- **Testing**: 9/10 — APPROVED
- **Architecture**: 8/10 — APPROVED
- **Complexity**: 9/10 — APPROVED
- **Performance**: 10/10 — APPROVED
- **Regression**: 10/10 — APPROVED
- **Reliability**: 9/10 — APPROVED
- **TypeScript**: 9/10 — APPROVED

### Status: READY FOR MERGE
All blocking ≥80% confidence findings have been documented via inline comments. The PR is architecturally sound, regression-clean, and backward-compatible. Address the inline comments before merge; lower-confidence suggestions are optional refinements.

---
*Comments created by Claude Code review automation*
*Confidence threshold: ≥80% for inline comments, 60-79% for summary recommendations*

File: src/cli/plugins.ts (hunk 1 above)
Comment (dean0x) [reply src/cli/plugins.ts]: Bare is deliberate here — this list is a **deletion manifest for stale paths**, not install names, and the prefix would invert its meaning:

- Init's legacy cleanup rm's `~/.claude/skills/<entry>` **verbatim, no prefix applied** (`init.ts:1027`). Bare `compliance` → the pre-namespace path `~/.claude/skills/compliance/` (a no-op today, disjoint from the live install).
- `devflow:compliance` here would point that same cleanup at `~/.claude/skills/devflow:compliance/` — the **live install dir** — and delete the skill on every `devflow init` (cleanup runs *after* install). Two tests pin this: `skill-namespace.test.ts:83` requires the bare form, and `plugins.test.ts:312` ("no namespaced legacy entry matches an active install path") fails on the prefixed form for exactly this reason.
- Shadowing is unaffected either way: shadow sources live under `~/.devflow/skills/{bare-name}/` (bare by convention, `installer.ts:365`), and legacy cleanup never touches `~/.devflow/` at all. Namespaced installation is applied by the installer (`prefixSkillName` → `~/.claude/skills/devflow:compliance/`), independent of this list.

The old comment ("bare name for pre-namespace installs") was the confusing part — compliance never shipped pre-namespace (namespacing landed 2026-03-30; compliance was born 2026-07-18), so the entry exists only to satisfy the `skill-namespace.test.ts:83` invariant. Reworded the comment to state the actual mechanism in 8bc8be8.

One genuine follow-up this surfaced (out of scope here): for skills born *after* namespacing, these bare entries have zero migration value but make init's unguarded `fs.rm` willing to delete a same-named **foreign** skill dir (`compliance` is a generic name). Fixing that means changing the `skill-namespace.test.ts:83` invariant to a frozen pre-namespace list and/or adding a devflow-marker guard to the cleanup — happy to do that as a follow-up PR if you want it.

Comment (dean0x) [reply src/cli/plugins.ts]: Follow-up: we took this one step further in 37f0712. You were right that no pre-namespace `compliance` install ever existed — the only thing forcing the entry was the test invariant ("every current skill must have a bare legacy entry"), which over-applied to skills born after namespacing. The invariant is now frozen to the 39 skills that actually existed before `dcecda3` (2026-03-30), with an inverse guard: a bare legacy entry for any post-namespace skill now **fails the test** (no migration value, only foreign-dir deletion risk). That pruned 30 vestigial bare entries (including `compliance`); all `devflow:`-prefixed entries and all genuine pre-namespace names (including old constituent names like `git-safety`/`architecture-patterns`) are untouched. Full suite green: 1,812 tests, tsc clean.
